File: JS/assignment4/PublicApi.DTO.v1/Todo/TodoCategory.cs
using System.ComponentModel.DataAnnotations;

namespace PublicApi.DTO.v1.Todo;

public class TodoCategory
{
    public Guid Id { get; set; }
        
    [MaxLength(128)]
    public string CategoryName { get; set; } = default!;

    public int CategorySort { get; set; }

    public DateTime SyncDt { get; set; } = DateTime.UtcNow;

    public string? Tag { get; set; }
}

File: JS/assignment6/src/views/todoCategory.ts
import type { TodoCategoryRow } from "../models/todoCategory.js";

export interface TodoCategoryDto {
  id: string;
  categoryName: string | null;
  categorySort: number;
  tag: string | null;
  syncDt: string;
}

export function categoryView(row: TodoCategoryRow): TodoCategoryDto {
  return {
    id: row.id,
    categoryName: row.categoryName,
    categorySort: row.categorySort,
    tag: row.tag,
    syncDt: row.syncDt,
  };
}
